packages/base/src/dialogs/symbology/symbologyUtils.ts

```diff
@@ -31,19 +32,27 @@ export namespace VectorUtils {
           // Third is input value that stop values are compared with
           // Fourth and on is value:color pairs
           for (let i = 3; i < color[key].length; i += 2) {
-            valueColorPairs.push({
-              stop: color[key][i],
-              output: color[key][i + 1]
-            });
+            const pairKey = `${color[key][i]}-${color[key][i + 1]}`;
+            if (!seenPairs.has(pairKey)) {
```

Comment:
I'm a bit worried about this approach, looks a lot like you're fixing the symptom and not the problem itself. Are the pair/value duplicated in `layer.parameters?.color`?

Reply:
> Are the pair/value duplicated in layer.parameters?.color?

No, they were duplicating in this method only as in https://github.com/geojupyter/jupytergis/pull/672, I changed from switch case using a condition to a for loop using a list iterating for both `['fill-color', 'circle-fill-color']` resulting in duplication

Reply:
In which case do we have that `pairKey` already there? It's when you have a duplication in `layer.parameters?.color`, right? That's what I understand from the code at least.

Reply:
We have that `pairKey` already there during the second iteration of for loop  for `'circle-fill-color'` after the first iteration of for loop for `'fill-color'`

I cannot see the duplication in `layer.parameters?.color`

Reply:
Ok got it!

I'm wondering how much we could simplify the schema itself to only have `color` steps and `radius` steps, not dealing with open-layers implementation details about `circle-fill-color` and `fill-color` at all in the schema. 

Not part of this PR though. Thanks!

Reply:
> I'm wondering how much we could simplify the schema itself to only have color steps and radius steps, not dealing with open-layers implementation details about circle-fill-color and fill-color at all in the schema.

If we do this, the schema would be closer to what we see in the Symbology panel, and this code would be simplified quite a lot.